addons/account/models/account_move.py

```diff
@@ -4752,6 +4765,17 @@ def write(self, vals):
 
         return result
 
+    @api.model
+    def new(self, values, origin=None, ref=None):
+        # If the line is created, with an invoice_id set for example but has no line_type in the vals or the context,
+        # make sure to set it to the default 'invl'
+        if not values:
+            values = {}
+        if 'line_type' not in values:
+            values['line_type'] = 'invl'
```

Comment:
hmmmm I don't understand why you don't get the right line_type here: to check with you

Reply:
Can happen when copying purchase line for example:

```python
        # Copy purchase lines.
        po_lines = self.purchase_id.order_line - self.line_ids.mapped('purchase_line_id')
        new_lines = self.env['account.move.line']
        for line in po_lines.filtered(lambda l: not l.display_type):
            new_line = new_lines.new(line._prepare_account_move_line(self))
            new_line.account_id = new_line._get_computed_account()
            new_line._onchange_price_subtotal()
            new_lines += new_line
```

As we use new here to create it, it does not set the line_type automatically. Of course, I could change _prepare_account_move_line to return 'invl' in line_type by default but I thought it makes it easier to centralize it in accounting.
I went by the principle that we should require a value on line_type only if something special is needed, otherwise it should be handled with default values in account.move. Let me know if it's wrong.

It is commented now, you can see some tests fail because the type is not properly set.